Assets/SEE/Game/Worlds/PlayerSpawner.cs

```diff
@@ -56,68 +49,75 @@ private void OnEnable()
-            {
-                Debug.LogError("Network configuration not found");
-                yield return null;
-            }
+            Net.Network networkConfig = FindObjectOfType<Net.Network>()
+                ?? throw new Exception("Network configuration not found.\n");
 
-            while (ReferenceEquals(networkManager, null))
+            // Wait until Dissonance is created.
+            while (ReferenceEquals(dissonanceComms, null))
             {
-                networkManager = NetworkManager.Singleton;
-                yield return null;
+                dissonanceComms = FindObjectOfType<DissonanceComms>();
+                if (ReferenceEquals(dissonanceComms, null))
+                {
+                    yield return null;
+                }
+                // We need to set the local player name in DissonanceComms
+                // before Dissonance is started. That is why we cannot afford
+                // to wait until the next frame.
+                dissonanceComms.LocalPlayerName = networkConfig.PlayerName;
             }
 
-            // Terminate this co-routine if not run by the server, but first set client playername for chat-function.
+            NetworkManager networkManager = NetworkManager.Singleton;
+
+            // Terminate this co-routine if not run by the server.
             if (!networkManager.IsServer)
             {
-                dissonanceComms = FindObjectOfType<DissonanceComms>();
-                dissonanceComms.LocalPlayerName = networkConfig.Playername;
                 yield break;
             }
 
             // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
             // The following code will be executed only on the server.
             // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
-            // Wait until Dissonance is created
-            while (ReferenceEquals(dissonanceComms, null))
-            {
-                dissonanceComms = FindObjectOfType<DissonanceComms>();
-                dissonanceComms.LocalPlayerName = networkConfig.Playername;
-                yield return null;
-            }
-
             // The callback to invoke once a client connects. This callback is only
             // ran on the server and on the local client that connects. We want
             // to spawn a player whenever a client connects.
-            networkManager.OnClientConnectedCallback += Spawn;
+            networkManager.OnClientConnectedCallback += ClientConnects;
             networkManager.OnClientDisconnectCallback += ClientDisconnects;
             // Spawn the local player. This code is executed by the server.
-            Spawn(networkManager.LocalClientId);
+            ClientConnects(networkManager.LocalClientId);
         }
 
         /// <summary>
         /// The number of players spawned so far.
         /// </summary>
         private int numberOfSpawnedPlayers = 0;
 
+        private static void Log(string message)
```

Comment:
Currently, we are surrounding every call of this method with an `#if DEBUG`. Instead, [Unity documentation recommends](https://docs.unity3d.com/Manual/UnderstandingPerformanceGeneralOptimizations.html) using the `Conditional` attribute on this `Log` method. That way, we don't have to wrap every call with these preprocessor directives and only have to do it once.

Reply:
Cool feature. Thanks for the hint. It's used now.